jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DownloadObjectAction.kt

```diff
@@ -32,7 +33,13 @@ import java.nio.file.Paths
 class DownloadObjectAction constructor(private val project: Project, treeTable: S3TreeTable) :
     S3ObjectAction(treeTable, message("s3.download.object.action"), AllIcons.Actions.Download) {
 
-    private data class DownloadInfo(val s3Object: String, val diskLocation: Path)
+    private data class DownloadInfo(val s3Object: String, val diskLocation: Path, val versionId: String = "") {
+        constructor(s3Object: S3TreeObjectNode, diskLocation: Path) : this(
+            s3Object.key,
+            diskLocation,
+            if (s3Object is S3TreeObjectVersionNode) s3Object.versionId else ""
```

Comment:
prefer null over empty string if there is no version id

Reply:
Right with refactoring of download method we don't need if else block so that's all good now.